index.js

```diff
@@ -275,33 +286,49 @@ class Footer extends React.Component {
     super(props);
   }
 
-  handleClick(e, state) {
-    const current = e.target;
-    current.classList.add('selected');
-    const parent = e.target.parentNode;
-    const siblings = [].slice.call(parent.children).filter(function(child) {
-      return child !== current;
-    });
-    siblings.forEach(sibling => sibling.classList.remove('selected'));
-    this.props.appState(state);
-  }
-
   render() {
-    const activeItemCount = this.props.items.filter(item => !item.isCompleted).length;
-    const completedItemCount = this.props.items.filter(item => item.isCompleted).length;
     return (
       <footer>
-        <p>{activeItemCount} {activeItemCount > 1 ? "items" : "item"} left</p>
-        {/*Create Enums for ApplicationState?*/}
-        <div className="filters">
-          <button id="all" className="selected" onClick={(e) => this.handleClick(e, "all")}>All</button>
-          <button id="active" onClick={(e) => this.handleClick(e, "onlyActive")}>Active</button>
-          <button id="completed" onClick={(e) => this.handleClick(e, "onlyCompleted")}>Completed</button>
-        </div>
-        {completedItemCount > 0 && <button id="clear-all" onClick={this.props.clearCompleted.bind(this)} >Clear completed</button>}
+        <p>{this.props.activeItemCount} {this.props.activeItemCount > 1 ? "items" : "item"} left</p>
+        <Filters>
+          <FilterButton filterName={APP_ALL} filter={this.props.filter} handleClick={(e) => this.props.handleFilter(APP_ALL, e)}/>
+          <FilterButton filterName={APP_ACTIVE} filter={this.props.filter} handleClick={(e) => this.props.handleFilter(APP_ACTIVE, e)}/>
+          <FilterButton filterName={APP_COMPLETED} filter={this.props.filter} handleClick={(e) => this.props.handleFilter(APP_COMPLETED, e)}/>
+        </Filters>
+        {this.props.completedItemCount > 0 && <button id="clear-all" onClick={this.props.clearCompleted} >Clear completed</button>}
       </footer>
     );
   }
 }
 
+class Filters extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    return <div className="filters">{this.props.children}</div>
+  }
+}
+
+class FilterButton extends React.Component {
+  constructor(props) {
+    super(props);
+  }
+
+  capitalize(filterName) {
+    return filterName.charAt(0).toUpperCase() + filterName.slice(1)
+  }
```

Comment:
**Nitpick:** rename `filterName` to something more generic, e.g.`str`.

As this is more like a generic string utility function it's better not to use any specific name for the variable (e.g. filterName).